ArduSub/Parameters.cpp

```diff
@@ -896,3 +896,152 @@ void Sub::convert_old_parameters()
+            JSButton::button_function_t::k_servo_1_max_toggle
+        },
+        {
+            // Actuator 2 (was servo_2_*)
+            JSButton::button_function_t::k_servo_2_inc,
+            JSButton::button_function_t::k_servo_2_dec,
+            JSButton::button_function_t::k_servo_2_min,
+            JSButton::button_function_t::k_servo_2_max,
+            JSButton::button_function_t::k_servo_2_center,
+            JSButton::button_function_t::k_servo_2_min_momentary,
+            JSButton::button_function_t::k_servo_2_max_momentary,
+            JSButton::button_function_t::k_servo_2_min_toggle,
+            JSButton::button_function_t::k_servo_2_max_toggle
+        },
+        {
+            // Actuator 3 (was servo_3_*)
+            JSButton::button_function_t::k_servo_3_inc,
+            JSButton::button_function_t::k_servo_3_dec,
+            JSButton::button_function_t::k_servo_3_min,
+            JSButton::button_function_t::k_servo_3_max,
+            JSButton::button_function_t::k_servo_3_center,
+            JSButton::button_function_t::k_servo_3_min_momentary,
+            JSButton::button_function_t::k_servo_3_max_momentary,
+            JSButton::button_function_t::k_servo_3_min_toggle,
+            JSButton::button_function_t::k_servo_3_max_toggle
+        }
+    };
+
+    // Target actuator functions for assignment
+    static constexpr SRV_Channel::Function actuator_functions[NUM_LEGACY_ACTUATORS] = {
+        SRV_Channel::Function::k_actuator1,
+        SRV_Channel::Function::k_actuator2,
+        SRV_Channel::Function::k_actuator3
+    };
+
+    // Process legacy channels 9-11
+    for (uint8_t actuator_idx = 0; actuator_idx < NUM_LEGACY_ACTUATORS; actuator_idx++) {
+        const uint8_t channel = FIRST_LEGACY_CHANNEL + actuator_idx;
+        const auto target_function = actuator_functions[actuator_idx];
+
+        // Skip if actuator function already assigned to any channel
+        uint8_t existing_channel;
+        if (SRV_Channels::find_channel(target_function, existing_channel)) {
+            continue;
+        }
+
+        // Skip if channel is not disabled
+        if (SRV_Channels::channel_function(channel - 1) != SRV_Channel::Function::k_none) {
+            continue;
+        }
+
+        // Check if any servo/actuator buttons are assigned
+        bool has_assigned_button = false;
+        for (size_t func_idx = 0; func_idx < FUNCTIONS_PER_SERVO; func_idx++) {
+            if (sub.jsbutton_function_is_assigned(servo_functions[actuator_idx][func_idx])) {
+                has_assigned_button = true;
+                break;
+            }
+        }
+
+        if (has_assigned_button) {
+            // Assign actuator function to preserve legacy behavior
+            char param_name[20];
+            snprintf(param_name, sizeof(param_name), "SERVO%u_FUNCTION", channel);
+            AP_Param::set_and_save_by_name(param_name, static_cast<int>(target_function));
```

Comment:
If there isn't an existing function to do this, it would be nice to refactor this out into a function so it can be reused (especially since you're using it twice already).